src/admin/example_middleware.erl

```diff
@@ -0,0 +1,55 @@
+%% Copyright (c) 2012-2025, DNSimple Corporation
+%%
+%% Permission to use, copy, modify, and/or distribute this software for any
+%% purpose with or without fee is hereby granted, provided that the above
+%% copyright notice and this permission notice appear in all copies.
+%%
+%% THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+%% WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+%% MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+%% ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+%% WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+%% ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+%% OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+%% @doc Example middleware for erldns admin API.
+%%
+%% This is a simple example middleware that demonstrates how to create
+%% custom middleware for the erldns admin API. This middleware adds a
+%% custom header to all responses.
```

Comment:
Using old edoc docs is not only handycaped but also totally redundant if you're also using the newer doc attributes as below. You can add this section as content to the `-moduledoc` attribute and remove the `@doc` comment entirely.